### PR TITLE
Harden determinism job

### PR DESCRIPTION
**Pull request by jaredpar**
This hardens the determinism job:
- Properly terminates build processes on exit
- Creates a zip file of all information needed for investigation on
failure

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure, @davkean

File: netci.groovy [hunk 1]
```diff
@@ -153,7 +153,7 @@ static void addStandardJob(def myJob, String jobName, String branchName, String
   addLogRotator(myJob)
   addWrappers(myJob)
 
-  def includePattern = "Binaries/**/*.pdb,Binaries/**/*.xml,Binaries/**/*.baseline*,Binaries/**/*.original*,,Binaries/**/*.key,Binaries/**/*.log,Binaries/**/*.dmp"
+  def includePattern = "Binaries/**/*.pdb,Binaries/**/*.xml,Binaries/**/*.log,Binaries/**/*.dmp,Binaries/**/*.zip"
```
Comment (davkean) [inline netci.groovy]: Was baseline/original the deterministic file?

**Comment by davkean** [comment]
:+1:

File: netci.groovy (hunk 1 above)
Comment (jaredpar) [inline netci.groovy]: Yes.  Now it's all in a zip though so no need to special case these entries.

**Pull request merged**